server/image_status_libpod.go

```diff
@@ -0,0 +1,67 @@
+// +build libpod
+
+package server
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	pb "k8s.io/kubernetes/pkg/kubelet/apis/cri/runtime/v1alpha2"
+)
+
+// ImageStatus returns the status of the image.
+func (s *Server) ImageStatus(ctx context.Context, req *pb.ImageStatusRequest) (resp *pb.ImageStatusResponse, err error) {
+	const operation = "image_status"
+	defer func() {
+		recordOperation(operation, time.Now())
+		recordError(operation, err)
+	}()
+
+	logrus.Debugf("ImageStatusRequest: %+v", req)
+
+	image, err := s.ImageRuntime().NewFromLocal(req.Image.Image)
+	if err != nil {
+		return nil, err
+	}
+
+	// Discarding the err here as I dont believe not getting size should be fatal
+	size, _ := image.Size(getContext())
+	resp = &pb.ImageStatusResponse{
+		Image: &pb.Image{
+			Id:          image.ID(),
+			RepoTags:    image.Names(),
+			RepoDigests: image.RepoDigests(),
+			Size_:       *size,
+		},
+	}
+
+	uid, username := getUserFromImage(image.User)
+	if uid != nil {
+		resp.Image.Uid = &pb.Int64Value{Value: *uid}
+	}
+	resp.Image.Username = username
+	logrus.Debugf("ImageStatusResponse: %+v", resp)
+	return resp, nil
+}
+
+// getUserFromImage gets uid or user name of the image user.
+// If user is numeric, it will be treated as uid; or else, it is treated as user name.
+func getUserFromImage(user string) (*int64, string) {
```

Comment:
Should libpod supply this helper?

Reply:
For now I think so ... that way I dot alter any of the non-libpod code track.

Reply:
> For now I think so ... that way I dont alter any of the non-libpod code track.

No, I meant "should we file a libpod PR adding this functionality to libpod's public API?".  It doesn't seem CRI-O specific, and I expect other folks consuming libpod as a library would want access to it as well.